addons/l10n_latam_check/models/account_payment.py

```diff
@@ -1,328 +1,252 @@
+                  "Checks:\n%s", ('\n'.join(['* %s (%s)' % (x.name, x.issue_state) for x in checks_reconciled])))
+            )
+
+    def action_cancel(self):
+        self._get_reconciled_checks_error()
+        super().action_cancel()
+
+    def action_draft(self):
+        self._get_reconciled_checks_error()
+        super().action_draft()
+
+    def _l10n_latam_check_split_move(self):
+        for payment in self.filtered(lambda x: x.payment_method_code == 'own_checks' and x.payment_type == 'outbound'):
+            if len(payment.l10n_latam_new_check_ids) == 1:
+                liquidity_line = payment._seek_for_lines()[0]
+                payment.l10n_latam_new_check_ids.liquidity_line_id = liquidity_line.id
+                continue
+            move_id = self.env['account.move'].create({
+                'journal_id': payment.journal_id.id,
+                'move_type': 'entry',
+            })
+            liquidity_line = payment._seek_for_lines()[0]
+
+            for check in payment.l10n_latam_new_check_ids:
+                liquidity_amount_currency = -check.amount
+                liquidity_balance = payment.currency_id._convert(
+                    liquidity_amount_currency,
+                    payment.company_id.currency_id,
+                    payment.company_id,
+                    payment.date,
+                )
+                # Checks liquidity line
+                move_line = self.env['account.move.line'].with_context(check_move_validity=False).create({
+                    'name': _('Check %s - ', check.name + liquidity_line.name),
+                    'date_maturity': check.payment_date,
+                    'amount_currency': liquidity_amount_currency,
+                    'currency_id': check.currency_id.id,
+                    'debit': max(0.0, liquidity_balance),
+                    'credit': -min(liquidity_balance, 0.0),
+                    'partner_id': payment.partner_id.id,
+                    'account_id': payment.outstanding_account_id.id,
+                    'move_id': move_id.id,
+                })
+                check.liquidity_line_id = move_line.id
+
+            # inverse liquidity line
+            inverse_liquidity_line = self.env['account.move.line'].with_context(check_move_validity=False).create({
+                    'name': liquidity_line.name,
+                    'date_maturity': liquidity_line.date_maturity,
+                    'amount_currency': -liquidity_line.amount_currency,
+                    'currency_id': liquidity_line.currency_id.id,
+                    'debit': -liquidity_line.debit,
+                    'credit': -liquidity_line.credit,
+                    'partner_id': payment.partner_id.id,
+                    'account_id': payment.outstanding_account_id.id,
+                    'move_id': move_id.id,
+                })
+            move_id.action_post()
+            (inverse_liquidity_line + liquidity_line).reconcile()
```

Comment:
Naming of the variables is confusing here.

Reply:
Done